src/dbnode/encoding/iterators.go

```diff
@@ -38,21 +39,71 @@ var (
 // iterators is a collection of iterators, and allows for reading in order values
 // from the underlying iterators that are separately in order themselves.
 type iterators struct {
-	values      []Iterator
-	earliest    Iterator
-	earliestIdx int
-	earliestAt  time.Time
-	filtering   bool
-	filterStart time.Time
-	filterEnd   time.Time
+	values             []Iterator
+	earliest           []Iterator
+	earliestAt         time.Time
+	filterStart        time.Time
+	filterEnd          time.Time
+	filtering          bool
+	equalTimesStrategy IterateEqualTimestampStrategy
+
+	// Used for caching reuse of value frequency lookup
+	valueFrequencies map[float64]int
 }
 
 func (i *iterators) len() int {
 	return len(i.values)
 }
 
 func (i *iterators) current() (ts.Datapoint, xtime.Unit, ts.Annotation) {
-	return i.earliest.Current()
+	numIters := len(i.earliest)
+
+	switch i.equalTimesStrategy {
+	case IterateHighestValue:
+		sort.Slice(i.earliest, func(a, b int) bool {
```

Comment:
Is this gonna alloc?

Reply:
Thankfully no, output of `go build -gcflags "-m" .` in this package:
```
./iterators.go:63:26: (*iterators).current func literal does not escape
./iterators.go:70:26: (*iterators).current func literal does not escape
./iterators.go:87:26: (*iterators).current func literal does not escape
```